File: LeetCode/Easy/1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        is_one = True
        ret = float("inf")
        temp = 0
        
        for v in s:
            if is_one and v == "1":
                temp += 1
            elif not is_one and v == "0":
                temp += 1
            is_one = not is_one
        
        ret = min(ret, temp)
        
        is_one = False
        temp = 0
        
        for v in s:
            if is_one and v == "1":
                temp += 1
            elif not is_one and v == "0":
                temp += 1
            is_one = not is_one
        
        ret = min(ret, temp)
        
        return ret
            

File: LeetCode/Easy/1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        char c1 = '1';
        char c2 = '0';
        int t1 = 0;
        int t2 = 0;
        
        for (char v : s){
            if (c1 == v) ++t1;
            else ++t2;
            
            char temp = c1;
            c1 = c2;
            c2 = temp;
        }
        
        return min(t1, t2);
    }
};